frontend/src/validation/index.js

```diff
@@ -23,6 +24,147 @@ const searchSchema = Yup.object().shape(
   ]
 )
 
+// Options for Visit Schema:
+// Strict doesn't coerce any values
+// Abort Early bails out at the first sign of error
+// Strip unknown removes unknown values, set to false because we don't expect any unknown values
+const options = {
+  strict: true,
+  abortEarly: false,
+  stripUnknown: false,
+}
+
+const visitSchema = Yup.object().shape({
+  id: Yup.number()
+    .defined()
+    .positive()
+    .integer(),
+  participant: Yup.number().when("id", {
+    is: null,
+    then: Yup.number().defined(),
+    otherwise: Yup.number()
+      .required()
+      .positive()
+      .integer(),
+  }),
+  program: Yup.number()
+    .required()
+    .positive()
+    .integer()
+    .min(1)
+    .max(10),
+  service: Yup.number()
+    .when("program", {
```

Comment:
This was a lot of work I see, but I'm not sure it's a good idea. I would rather not hardcode the database ids into the front end.  both of these things, and especially services, are subject to addition and removal.  If we really wanted to check this, can we create a helper function that's similar to how the services are filtered for selection? it might be overkill, the backend will deny a mismatch anyway.

Reply:
So, yeah I kinda knew this was overkill. I'm going to remove it. I can look into adding a helper function to save us a trip to the backend to validate.